mne/io/base.py

```diff
@@ -2549,6 +2549,12 @@ def concatenate_raws(raws, preload=None, events_list=None, *,
     events : ndarray of int, shape (n_events, 3)
         The events. Only returned if ``event_list`` is not None.
     """
+    # consistent channel order across raws (not checked by _ensure_infos_match)
+    ch_order = raws[0].ch_names
+    for raw in raws[1:]:
+        raw.reorder_channels(ch_order)
+    # [raw.reorder_channels(raws[0].ch_names) for raw in raws[1:]]
```

Comment:
This is simple, but it would reorder the `raw[1:]` in place, modifying the user variable outside the scope of this function. I think you need to check one by one if the order is different and then **copy** and re-order only the ones that have a different order.

Reply:
proposal: can we make it so that `_ensure_infos_match` *does* check channel order (thereby making `concatenate_raws` fail if they don't) and then provide a new function (`match_channel_orders` or so) that we advise users to run before concatenating?

Reply:
I would mimic the api of concatenation of epochs or the grand average function for evoked

Reply:
@moritz-gerster can you comment here?

Reply:
@larsoner I don't understand, what should I comment? I posted two comments right above your comment.

So far, I implemented a solution that requires further discussion and implemented a test that passes under the current implementation. I am waiting for all of you to agree before I proceed. Currently, @mscheltienne, @drammock, and @agramfort suggested three very different solutions. I am not experienced enough to choose one of them.